src/main/java/reposense/authorship/model/FileResult.java

```diff
@@ -14,34 +14,42 @@
     private final String path;
     private FileType fileType;
     private Boolean isBinary = null; // Should only be true or null to prevent it from being serialized
+    private Boolean isIgnored = null;
     private final ArrayList<LineInfo> lines;
     private final HashMap<Author, Integer> authorContributionMap;
 
     public FileResult(String path, FileType fileType, ArrayList<LineInfo> lines,
-            HashMap<Author, Integer> authorContributionMap, boolean isBinary) {
+            HashMap<Author, Integer> authorContributionMap, boolean isBinary, boolean isIgnored) {
         this.path = path;
         this.fileType = fileType;
         this.lines = lines;
         this.authorContributionMap = authorContributionMap;
+        // isBinary and isIgnored are mutually exclusive
         if (isBinary) {
             this.isBinary = true;
+        } else if (isIgnored) {
+            this.isIgnored = true;
         }
```

Comment:
The else case here would be valid because most files would be neither binary not ignored. I've added an assertion that `isBinary` and `isIgnored` are not both true, since that would indicate programmer error that should be fixed.